### DISCUSSION TITLE
FNIRSI GC-01: Issue when returning to original firmware

### DISCUSSION THREAD

**Discussion by alfmck** [Gissio/radpro#52] [Q&A] [answered]
After going back to original flash on GC-01 with Geehy CH32F103R8T6 MPU (no matter v0.3 or v1.5), there is no text on the light grey background. It seems like letters also are light grey. In GC-01 with WCH MPU there isn't such issue.
![IMG_3176](https://github.com/Gissio/radpro/assets/165254519/ac86cb49-a240-4953-a3ae-8cdd59ac0377)
But everything else is working OK. Does somebody also have this problem?

**Comment by ricsim** [comment]
Yes, I had exactly the same.
I flashed with the ST-Link Flasher, as I cannot get the Drive to show up. (Windows gives me error, and Linux shows the drive, but it cannot flash)

**Reply by alfmck** [reply, marked as answer]
And it's a pity case. I've written this post to warn others.
Linux or win7 cannot flash, but I'm flashing anyway in order to "kill" old soft. After that, the newest win11 opens the Drive, but in open window you see ERROR instead of READY. Then I flash new soft, and everything works.

**Comment by gc01noob** [comment]
to flash the GC01:
1. use Linux
2. open the BOOTLOADER drive
3. drag over the new firmware .bin
4. leave the window open
5. drag over the firmware .bin for the second time
6. OK the overwrite.
7. viola it boots with the new firmware and you can disconnect.

**Reply by gc01noob** [reply]
works for upgrade and downgrade (using any USB2 or 3 ports)
you can even live boot Linux if you don't have it installed - I used Ubuntu 22.04 for simplcity

forget Windows 10/11 for flashing, just use it to run GeigerLog

**Reply by dc1rdb** [reply]
Works with Debian Buster on a Raspberry Pi, too :-)

**Comment by zhang00963** [comment]
The firmware is encrypted because the bootloader contains the encrypted information of the chip, and it is one-to-one encrypted. When the bootloader in the warehouse is flashed, it will look like this without displaying text prompts, but I have already solved it.
